File: cpp-with-qt/exampletest.cpp
/* Since unit test class, only include if TEST macro defined */
#ifdef TEST

#include "exampletest.h"
#include <QString>


void TestExampleClass::whatsTheWordTest() {
  const QString theWord = example.whatsTheWord();
  QVERIFY("bird" == theWord);
}

void TestExampleClass::isTheWordCorrect() {
  bool result = example.isTheWord(QString("bird"));
  QVERIFY(result);
}

void TestExampleClass::isTheWordCorrectCasesAndTrim() {
  bool result = example.isTheWord(QString("  BIrD  "));
  QVERIFY(result);
}

void TestExampleClass::isTheWordInCorrect() {
  bool result = example.isTheWord(QString("not the bird"));
  QVERIFY(!result);
}

#endif

File: csharp-with-vs/ubb-se-example-csharp-with-vs-test/ExampleTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using UbbSeExample;

namespace UbbSeExampleTest
{
  /**
   * A few unit tests for ExampleClass.
   * @author Csaba Sulyok
   */
  [TestClass]
  public class ExampleTest
  {
    [TestMethod]
    public void WhatsTheWordTest() {
      Example example = new Example();
      string theWord = example.WhatsTheWord();
      Assert.IsTrue("bifrd" == theWord);
    }

    [TestMethod]
    public void IsTheWordCorrect() {
      Example example = new Example();
      bool result = example.IsTheWord("bird");
      Assert.IsTrue(result);
    }

    [TestMethod]
    public void IsTheWordCorrectCasesAndTrim() {
      Example example = new Example();
      bool result = example.IsTheWord("  BIrD  ");
      Assert.IsTrue(result);
    }

    [TestMethod]
    public void IsTheWordInCorrect() {
      Example example = new Example();
      bool result = example.IsTheWord("not the bird");
      Assert.IsFalse(result);
    }
  }
}
